yarn-project/sequencer-client/src/config.ts

```diff
@@ -90,9 +98,87 @@ export function getConfigEnvVars(): SequencerClientConfig {
     feeRecipient: FEE_RECIPIENT ? AztecAddress.fromString(FEE_RECIPIENT) : undefined,
     acvmWorkingDirectory: ACVM_WORKING_DIRECTORY ? ACVM_WORKING_DIRECTORY : undefined,
     acvmBinaryPath: ACVM_BINARY_PATH ? ACVM_BINARY_PATH : undefined,
-    allowedFeePaymentContractClasses: SEQ_FPC_CLASSES ? SEQ_FPC_CLASSES.split(',').map(Fr.fromString) : [],
-    allowedFeePaymentContractInstances: SEQ_FPC_INSTANCES
-      ? SEQ_FPC_INSTANCES.split(',').map(AztecAddress.fromString)
-      : [],
+    allowedFunctionsInSetup: SEQ_ALLOWED_SETUP_FN
+      ? parseSequencerAllowList(SEQ_ALLOWED_SETUP_FN)
+      : getDefaultAllowedSetupFunctions(),
+    allowedFunctionsInTeardown: SEQ_ALLOWED_TEARDOWN_FN
+      ? parseSequencerAllowList(SEQ_ALLOWED_TEARDOWN_FN)
+      : getDefaultAllowedTeardownFunctions(),
   };
 }
+
+function parseSequencerAllowList(value: string): AllowedFunction[] {
+  const entries: AllowedFunction[] = [];
+
+  if (!value) {
+    return entries;
+  }
+
+  for (const val of value.split(',')) {
+    const [identifierString, selectorString] = val.split(':');
+    const selector = FunctionSelector.fromString(selectorString);
+
+    if (identifierString.startsWith('0x')) {
```

Comment:
Not the best parsing since it assumes that an `AztecAddress extends Fr` but it should always be prefixed with `0x`